scripts/ci/check_compliance.py

```diff
@@ -1541,20 +1541,26 @@ class KeepSorted(ComplianceTest):
 
     MARKER = "zephyr-keep-sorted"
 
-    def block_is_sorted(self, block_data):
-        lines = []
+    def block_check_sorted(self, block_data, regex):
+        if regex is None:
+            block_data = textwrap.dedent(block_data)
+
+        last = None
+        for idx, line in enumerate(block_data.splitlines()):
+            # check for regex
+            if regex and not re.match(regex, line):
+                continue
 
-        for line in textwrap.dedent(block_data).splitlines():
-            if len(lines) > 0 and line.startswith((" ", "\t")):
-                # Fold back indented lines
-                lines[-1] += line.strip()
-            else:
-                lines.append(line.strip())
+            # ignore indented lines
+            if not regex and line.startswith((" ", "\t")):
+                continue
```

Comment:
Hey, the foldback logic was to handle cases where the whole part to be sorted is part of the indeded stuff, like

```
zephyr_library_sources_ifdef(
        CONFIG_INPUT_ADC_KEYS input_adc_keys.c)
zephyr_library_sources_ifdef(
        CONFIG_INPUT_ANALOG_AXIS input_analog_axis.c)
```

does this still work here? I have the feeling that it won't

Reply:
Ah, I wondered why the lines were kept! Will fix.

Reply:
Added folding logic back.